### DISCUSSION TITLE
/stations limit stuck at 200

### DISCUSSION THREAD

**Discussion by brianolson** [weather-gov/api#234] [Q&A]
**Describe the bug**
`/stations?state=VT` only returns 200 items even if I as for 400.

**To Reproduce**
```sh
curl -H 'User-Agent: myappid' -o api_vt_stations.json 'https://api.weather.gov/stations?state=VT&limit=400'
grep -c stationIdentifier api_vt_stations.json
```
This shows me that I got 200 responses.
The docs at `https://api.weather.gov/openapi.json` indicate that the default limit is 200 with a max of 500. I'm guessing the limit is stuck at 200 and there don't just happen to be exactly 200 weather stations in Vermont.

**Expected behavior**
`&limit=400` should allow me to fetch 400 records.
**OR**
The stations list endpoint needs pagination. Maybe sort by stationIdentifier and let me query for stations greater than the last on I saw on my previous request.

**Environment**
`curl` or Python 3.8 on Linux

**Comment by JJKraw** [comment]
Are you sure that there are > 200 stations to be returned in VT? I just tried MA with no limit parameter and it comes back with 349. CA returns over 3500.

I'd like to know if there is documentation on where these stations come from (e.g. are they CWOP or something else) and also why when using the /gridpoints station list only returns a subset of these stations.

**Comment by StephenClouse** [comment]
I see about 350 stations in the database for VT. I'll have to look into what's going on here.

@JJKraw We pull the station metadata from MesoWest, which is pretty much everything that qualifies as a weather station.

The station lists returned for gridpoints are a list curated by the forecast office. These are the stations that will be exclusively used for displaying current conditions on point forecast pages on forecast.weather.gov, among other things.

**Comment by JJKraw** [comment]
Thanks. I was wondering why my station doesn't show up - maybe that's a related problem?

  https://mesowest.utah.edu/cgi-bin/droman/meso_base.cgi?stn=F4597

I get a 500 / unexpected problem if I query https://api.weather.gov/stations/F4597

(Not trying to hijack this thread - I'll start another if that's appropriate)

**Comment by StephenClouse** [comment]
@JJKraw I took a look and the issue is that station just isn't in the production database, because it apparently needs an update. We've been having trouble with regular updates.

Although that should definitely be a 404 rather than a 500. I'll look at fixing that.

**Comment by JJKraw** [comment]
Thanks a lot!! I appreciate you taking the time to look into it.
